core/server/src/main/java/alluxio/master/journal/ProtoBufJournalFormatter.java

```diff
@@ -40,11 +47,25 @@ public void serialize(JournalEntry entry, OutputStream outputStream) throws IOEx
   @Override
   public JournalInputStream deserialize(final InputStream inputStream) throws IOException {
     return new JournalInputStream() {
+      private final byte[] mBuffer = new byte[1024];
       private long mLatestSequenceNumber;
 
       @Override
       public JournalEntry getNextEntry() throws IOException {
-        JournalEntry entry = JournalEntry.parseDelimitedFrom(inputStream);
+        int firstByte = inputStream.read();
```

Comment:
It looks like you are parsing some of the proto manually. Could you briefly comment on the format?